File: src/c_tests/test_refvap.c
#include <stdlib.h>
#include <complex.h>
#include <stdio.h>

#include "../caobs_main/refvap.h"

int main(int argc, char *argv[]){
  
  double nu = 22000000000.0; // 22 GHz
  //double nu = 2.20000010E+10;
  double t = 293.0; // Roughly 20 Celcius
  double pvap = 2317.0;
  double pdry = 101325.0 - pvap;

  double complex nvap = refvap(nu, t, pdry, pvap);
  fprintf(stdout, "%.8E\n", nu);
  fprintf(stdout, "%f\n", t);
  fprintf(stdout, "%f\n", pdry);
  fprintf(stdout, "%f\n", pvap);
  fprintf(stdout, "(%f, %.8E)\n", creal(nvap), cimag(nvap));
  
  return EXIT_SUCCESS;
}

File: src/fortran_tests/test_refvap.f90
program test_refvap
  implicit none

  ! Set up a test case for an execution of refvap.
  COMPLEX refvap ! Function definition
  COMPLEX nvap ! Value coming from refvap

  ! Observing frequency in Hz, as a real
  REAL nu
  ! Atmospheric temperature in K, as a real
  REAL t
  ! Partial pressure of dry components in Pa, as a real
  REAL pdry
  ! Partial pressure of water vapour in Pa, as a real
  REAL pvap

  nu = 22000000000.0 ! 22 GHz
  t = 293.0 ! Roughly 20 Celcius
  pvap = 2317.0
  pdry = 101325.0 - pvap
  
  nvap = refvap(nu, t, pdry, pvap)
  print *, nu
  print *, t
  print *, pdry
  print *, pvap
  print *, nvap
  
end program test_refvap

